### PR TITLE
[WINC-496] [ci] Add east/west e2e tests through a clusterIP

### PR DESCRIPTION
**Pull request by sebsoto**
This commit increases the coverage of the east to west networking tests
to include reaching the Windows webserver through a clusterIP service.
This commit also accomplishes a refactoring of the existing tests into a
table driven test format.

Co-authored-by: Sebastian Soto <ssoto@redhat.com>

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1390,4 +1390,5 @@ sigs.k8s.io/yaml v1.1.0/go.mod h1:UJmg0vDUVViEyp3mgSv9WPwZCDxu4rQW1olrI1uml+o=
 sigs.k8s.io/yaml v1.2.0 h1:kr/MCeFWJWTwyaHoR9c8EjH9OumOmoF9YGiZd7lFm/Q=
 sigs.k8s.io/yaml v1.2.0/go.mod h1:yfXDCHCao9+ENCvLSE62v9VSji2MKu5jeNfTrofGhJc=
 vbom.ml/util v0.0.0-20160121211510-db5cfe13f5cc/go.mod h1:so/NYdZXCz+E3ZpW0uAoCj6uzU2+8OWDFv/HxUSs7kI=
+vbom.ml/util v0.0.0-20180919145318-efcd4e0f9787 h1:O69FD9pJA4WUZlEwYatBEEkRWKQ5cKodWpdKTrCS/iQ=
```
Comment (aravindhp) [inline go.sum:1393]: Why is this needed?

**Review by aravindhp — APPROVED** [review]
Thanks for the quick turnaround, @sebsoto 

/approve

File: test/e2e/network_test.go [hunk 2]
```diff
@@ -38,109 +37,108 @@ var (
 	retryInterval = 5 * time.Second
 )
 
-// testEastWestNetworking deploys Windows and Linux pods, and tests that the pods can communicate
-func testEastWestNetworking(t *testing.T) {
-	testCtx, err := NewTestContext(t)
-	require.NoError(t, err)
+// osTypes is used to specify an operating system to run workloads on
+type osTypes string
```
Comment (VaishnaviHire) [inline test/e2e/network_test.go:41]: I think this could be `operatingSystem`

**Comment by PratikMahajan** [comment]
LGTM

File: test/e2e/network_test.go [hunk 3]
```diff
@@ -38,109 +37,108 @@ var (
...
+			// So trying multiple times
+			var winServerDeployment *appsv1.Deployment
+			for i := 0; i < deploymentRetries; i++ {
+				winServerDeployment, err = testCtx.deployWindowsWebServer("win-webserver-"+strings.ToLower(node.Status.NodeInfo.MachineID), affinity)
+				if err == nil {
+					break
+				}
+			}
+			require.NoError(t, err, "could not create Windows Server deployment")
+			defer testCtx.deleteDeployment(winServerDeployment.Name)
```
Comment (VaishnaviHire) [inline test/e2e/network_test.go:99]: Is there a reason to move from deletion at the [end](https://github.com/openshift/windows-machine-config-operator/pull/129/files#diff-96873a210ce97729b66ec711eb21c852L79) to `defer` ?

File: go.sum (hunk 1 above)
Comment (sebsoto) [reply go.sum:1393]: im not sure. It came up from a go mod tidy

File: test/e2e/network_test.go [hunk 4]
```diff
@@ -480,15 +478,15 @@ func getWinCurlerCommand(winServerIP string) []string {
...
-func (tc *testContext) createWindowsServerJob(name string, command []string) (*batchv1.Job, error) {
+func (tc *testContext) createWindowsServerJob(name string, command []string, affinity *v1.Affinity) (*batchv1.Job, error) {
 	windowsNodeSelector := map[string]string{"beta.kubernetes.io/os": "windows"}
 	windowsTolerations := []v1.Toleration{{Key: "os", Value: "Windows", Effect: v1.TaintEffectNoSchedule}}
 	windowsServerImage := tc.getWindowsServerContainerImage()
-	return tc.createJob(name, windowsServerImage, command, windowsNodeSelector, windowsTolerations)
+	return tc.createJob(name, windowsServerImage, command, windowsNodeSelector, windowsTolerations, affinity)
 }
 
 func (tc *testContext) createJob(name, image string, command []string, selector map[string]string,
```
Comment (VaishnaviHire) [inline test/e2e/network_test.go:488]: Missing doc

**Review by VaishnaviHire — COMMENTED** [review]
Thanks @sebsoto for working on this. PTAL at my comments

File: test/e2e/network_test.go (hunk 2 above)
Comment (LorbusChris) [reply test/e2e/network_test.go:41]: ^ that, or at least singular

File: test/e2e/network_test.go (hunk 3 above)
Comment (sebsoto) [reply test/e2e/network_test.go:99]: Yes, this way we still clean up resources even when we fail the test

**Comment by sebsoto** [comment]
failures from cluster spinup

**Comment by sebsoto** [comment]
OVN cluster creation issues in the last two sets of failures
/retest
